src/content/lodash/string/kebabCase/vanilla.js

```diff
@@ -1,20 +1,53 @@
 const kebabCase = str =>
-  str
-    // Add hypen before capital letters followed by lowercase letters
-    .replace(/[A-Z][a-z]/g, match => `-${match}`)
-    // Everything that is not alphanumeric with an hypen
-    .replace(/[^A-Za-z0-9]/g, '-')
-    // Replace consecutive hyphens with a single hyphen
-    .replace(/-{2,}/g, '-')
-    // Remove hyphen from start and end of the string
-    .replace(/(^-|-$)/g, '')
-    .toLowerCase()
+    (String(str ?? "").match(/([0-9]+|([A-Z][a-z]+)|[a-z]+|([A-Z]+)(?![a-z]))/g) ?? [""])
+        .map((x) => x.toLowerCase())
+        .join("-");
 
-exports.first = kebabCase('Foo Bar')
+module.exports.fooSpaceBar = kebabCase('Foo Bar');
 // => 'foo-bar'
-
-exports.second = kebabCase('fooBar')
+module.exports.fooBar = kebabCase('fooBar');
 // => 'foo-bar'
-
-exports.third = kebabCase('__FOO_BAR__')
+module.exports.foo_bar =  kebabCase('__FOO_BAR__');
+// => 'foo-bar'
+module.exports.varFooBar =  kebabCase("FooBar");
```

Comment:
can you do the same here with the `module.exports.varname` to `exports.varname`? After that happy to merge

Reply:
Of course ^^